_DeploymentAndDistroPackaging/ansible/roles/ciao-controller/tasks/certificates.yml

```diff
@@ -48,6 +48,8 @@
         creates: cert-CNCIAgent-localhost.pem
       - role: controller
         creates: cert-Controller-localhost.pem
+      - role: agent,netagent
+        creates: cert-CNAgent-NetworkingAgent-localhost.pem
```

Comment:
There is no easy way to iterate the list of host and determine if there are dual nodes.
Also its better to have the cert creation in the same place since the cert is created once.

And lastly, while I agree that is a bit dirty to generate the file even if it is not going to be used... It doesnt mean it makes the deployment less secure. If an attacker somehows get access to that file, it will get access to the rest of the certificates anyways that are always generated.